.github/workflows/ci-pipeline.yml

```diff
@@ -0,0 +1,20 @@
+name: Continuous Integration
+on:
+  push:
+      paths-ignore:
+      - 'README.md'
+      - 'docs/**'
+      - 'docker-compose-*.yaml'
+  pull_request:
+    paths-ignore:
+      - 'README.md'
+      - 'docs/**'
+      - 'docker-compose-*.yaml'
+
+jobs:
+  build:
+    name: Build and test
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - run: docker build --target test --tag todo-app-tests . && docker run todo-app-tests
```

Comment:
Minor, but would this be nicer as two separate commands? Thoughts in my head basically revolve around _slightly_ quicker feedback on whether it was the building the image vs running the tests that failed (e.g. if this triggered a notification, knowing that it happened during build might be useful context)